### PR TITLE
[SPARK-40396][BUILD] Update scalatest and scalatestplus related dependencies to use stable version

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr update `scalatest` and `scalatestplus` related dependencies to use stable version as follows:

- Upgrade  `scala-xml` from 1.20 to `2.1.0` to support scalatest 3.2.13

- Update `org.scalatest:scalatest` from 3.3.0-SNAP3 to 3.2.13

- Update `org.scalatestplus:scalacheck-1-15:3.3.0-SNAP3` to `org.scalatestplus:scalacheck-1-16:3.2.13.0` and upgrade `scalacheck` from 1.15.4 to 1.16.0

- Update `org.scalatestplus:selenium-3-141:3.3.0.0-SNAP3` to `org.scalatestplus:selenium-3-141:3.2.10.0` and left TODO of SPARK-40397.

### Why are the changes needed?
Change to use stable version dependencies.

The relevant release notes as follows:
- scala-xml:
   -  https://github.com/scala/scala-xml/releases/tag/v1.3.0
   - https://github.com/scala/scala-xml/releases/tag/v2.0.0
   - https://github.com/scala/scala-xml/releases/tag/v2.0.1
   - https://github.com/scala/scala-xml/releases/tag/v2.1.0
- scalatest : 
   - https://github.com/scalatest/scalatest/releases/tag/release-3.2.10
   - https://github.com/scalatest/scalatest/releases/tag/release-3.2.11
   - https://github.com/scalatest/scalatest/releases/tag/release-3.2.12
   - https://github.com/scalatest/scalatest/releases/tag/release-3.2.13
- org.scalatestplus:scalacheck: 
   - https://github.com/scalatest/scalatestplus-scalacheck/releases/tag/release-3.2.10.0-for-scalacheck-1.15
   - https://github.com/scalatest/scalatestplus-scalacheck/releases/tag/release-3.2.11.0-for-scalacheck-1.15
   - https://github.com/scalatest/scalatestplus-scalacheck/releases/tag/release-3.2.12.0-for-scalacheck-1.16
   - https://github.com/scalatest/scalatestplus-scalacheck/releases/tag/release-3.2.13.0-for-scalacheck-1.16
- org.scalatestplus:mockito:
   - https://github.com/scalatest/scalatestplus-mockito/releases/tag/release-3.2.13.0-for-mockito-4.6 
- org.scalatestplus:selenium:
   - https://github.com/scalatest/scalatestplus-selenium/releases/tag/release-3.2.10.0-for-selenium-3.141

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?

- Pass GitHub Actions
- Manually check the problems mentioned in https://github.com/apache/spark/pull/35128/files

```
SPARK_GENERATE_GOLDEN_FILES=1 build/sbt clean "sql/testOnly *PlanStability*Suite"
[info] Run completed in 52 seconds, 520 milliseconds.
[info] Total number of tests run: 334
[info] Suites: completed 7, aborted 0
[info] Tests: succeeded 334, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1139,37 +1139,38 @@
       <dependency>
         <groupId>org.scalatest</groupId>
         <artifactId>scalatest_${scala.binary.version}</artifactId>
-        <version>3.3.0-SNAP3</version>
+        <version>3.2.13</version>
```
Comment (srowen) [inline pom.xml:1142]: This seems like a downgrade?

Comment (LuciferYang) [reply pom.xml:1142]: Not downgrade:

1. https://search.maven.org/artifact/org.scalatest/scalatest_2.12
image

The update time of `3.3.0-SNAP3` is 11-Dec-2020,  this is earlier than the release of 3.2.4

2. https://github.com/scalatest/scalatest/tags

There is no 3.3.x officially tag and it seems that the 3.3.x branch is still under development or has been abandoned(The latest commit was 16 Mar)

So I don't think we should use this version

Comment (srowen) [reply pom.xml:1142]: Later in time doesn't mean later version. If you're just saying this seems like a somewhat abandoned 3.3.x release, OK

File: pom.xml [hunk 2]
```diff
@@ -1109,7 +1109,7 @@
       <dependency>
         <groupId>org.scala-lang.modules</groupId>
         <artifactId>scala-xml_${scala.binary.version}</artifactId>
-        <version>1.2.0</version>
+        <version>2.1.0</version>
```
Comment (LuciferYang) [inline pom.xml:1112]: update this due to 

image

**Review by srowen — APPROVED** [review]

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @LuciferYang and @srowen for review.

**Comment by LuciferYang** [comment]
thanks @srowen and @MaxGekk

**Comment by pan3793** [comment]
branch-3.3 uses the abandoned version `3.3.0-SNAP3` now, I suggest we backport this PR to branch-3.3 cc @wangyum

**Comment by LuciferYang** [comment]
> branch-3.3 uses the abandoned version `3.3.0-SNAP3` now, I suggest we backport this PR to branch-3.3 cc @wangyum

This pr also upgrade scala-xml from 1.20 to 2.1.0, maybe we need try to find a version that doesn't need to upgrade `scala-xml`

**Pull request closed without merging**